router/worker.go

```diff
@@ -1070,13 +1070,23 @@ func (w *worker) sendDestinationResponseToConfigBackend(payload json.RawMessage,
 }
 
 func (w *worker) retryLimitReached(status *jobsdb.JobStatusT) bool {
+	respStatusCode, _ := strconv.Atoi(status.ErrorCode)
+	switch respStatusCode {
+	case types.RouterTimedOutStatusCode,
+		types.RouterUnMarshalErrorCode: // 5xx errors
```

Comment:
If there is a problem unmarshaling in the router then there is no point in retrying, right? So we can return `true`?

Reply:
What you say makes sense, but this is existing behaviour and I think there's merit to it.
I think it's this way because an unmarshal error means a breach of transformer<->server contract.
Aborting those jobs would only force us to replay in these cases.
Whereas letting them fail gives us time to fix the transformer/server issues causing this.
@rudderlabs/server_team wdyt?

Reply:
We also do have an alert if the unmarshal fails IIRC so we should let it be as is I guess

Reply:
OK so here we're saying that, since there is a problem with the transformer then we get an alert as Rohith said and while the messages are being retried we fix the bug with the transformer, create a release and deploy it live, hopefully all of this before the messages are discarded/expired.

Is this actually likely to happen? There are other reasons for this?